src/progressbar/progressbar.ts

```diff
@@ -0,0 +1,49 @@
+import {Component, Input} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-progressbar',
+  template: `
+    <progress class="progress {{striped && 'progress-striped'}} {{type && 'progress-' + type}}" [value]="value" [max]="max">
+      <div class="progress">
+        <span class="progress-bar" [style.width.%]="calculatePercentage()"><ng-content></ng-content></span>
+      </div>
+    </progress>
+  `
+})
+export class NgbProgressbar {
+  private _max = 100;
+  private _striped = false;
+  private _value: number;
+  min = 0;
+
+  @Input()
+  set max(value: number | string) {
+    this._max = parseInt(`${value}`, 10);
+    this.calculatePercentage();
```

Comment:
no need to call `calculatePercentage()` here - it is going to be called from a template